### DISCUSSION TITLE
Working with many URLs

### DISCUSSION THREAD

**Discussion by fthiery** [umami-software/umami#1052] [Q&A]
Hi, considering switching over to umami to replace GA.

The live demo only shows 19 URLs at https://app.umami.is/share/8rmHaheU/umami.is?view=url ; what happens if you have thousands of URLs ? 

- Are only the top 20 shown ?
- Is the display paginated ?
- Can stats about low traffic URLs still be accessed using the querystring (like `https://app.umami.is/share/8rmHaheU/umami.is?url=%2Fdocs%2Fcustom-url&view=country`) ?
- Our resources can be accessed through 2 URLs (one unique id and one slug-based), is there a way to "merge" these ?

Thanks for your feedback,

**Comment by mikecao** [comment]
That's not the top 20, those are all the urls. There is current no pagination but it's something we could implement if needed.

We will add more filtering features in the futures, like country, browser, device, etc.

You would have to disable automatic tracking and do it programmatically. See https://umami.is/docs/tracker-functions

**Comment by fthiery** [comment]
Thanks for your response. What happens currently if you have, say, 20000 urls ? All URLs are fetched at once (likely to timeout) ?

**Reply by mikecao** [reply]
Yes it will just fetch all the urls for the given timeframe.
